File: front/minhas-despesas/src/models/periodo-dto.ts
import { ContaDto } from "./conta-dto"
import { DespesaDto } from "./despesa-dto"

export interface PeriodoDto {
    periodoId: string,
    conta: ContaDto,
    contaId: string,
    mes: number,
    ano: number,
    despesas: Array<DespesaDto>,
    valor: number,
    valorPago: number,
    valorFalta: number
}

File: back/MinhasDespesas/MinhasDespesas.Models/Dtos/PeriodoDto.cs
using MinhasDespesas.Models.Entities;
using MinhasDespesas.Models.Enums;

namespace MinhasDespesas.Models.Dtos;

public class PeriodoDto
{
    public Guid PeriodoId { get; set; }
    public ContaDto Conta { get; set; }
    public Guid ContaId { get; set; }
    public Meses Mes { get; set; }
    public int Ano { get; set; }
    public IList<DespesaDto> Despesas { get; set; }=new List<DespesaDto>();
    public double Valor { get; set; }
    public double ValorPago { get; set; }
    public double ValorFalta { get; set; }
}